### PR TITLE
sim: or1k: Get building again

### PR DESCRIPTION
**Pull request by stffrdhrn**
The or1k sim here is was we want to push to upstream since it does not depend on or1ksim. 
I have enabled the sim build in travis since it should build now.

### CODE REVIEW

File: gdb/or1k-tdep.c [hunk 1]
```diff
@@ -463,7 +464,7 @@ or1k_return_value (struct gdbarch  *gdbarch,
 
       return RETURN_VALUE_ABI_RETURNS_ADDRESS;
     }
-
+
```
Comment (imphil) [inline gdb/or1k-tdep.c]: trailing whitespace

File: gdb/or1k-tdep.c [hunk 2]
```diff
@@ -439,12 +439,13 @@ or1k_return_value (struct gdbarch  *gdbarch,
...
 
-  /* Deal with struct/union first. If this won't fit in a single register it
-     is returned in memory. Large (2 word) scalars are returned in r11 and r12
-     (this is a change from GCC 4.2.2, when they were apparently returned in
-     memory). */
-  if (((TYPE_CODE_STRUCT == rv_type) || (TYPE_CODE_UNION  == rv_type)) &&
-      (rv_size >  bpw))
+  /* Deal with struct/union as addresses. If an array won't fit in a single 
+     register it is returned as address. Anything larger than 2 registers needs
+     to also be passed as address (this is maily from default_return_in_memory
```
Comment (imphil) [inline gdb/or1k-tdep.c]: just a drive-by: s/maily/mainly/

File: gdb/or1k-tdep.c (hunk 1 above)
Comment (stffrdhrn) [inline gdb/or1k-tdep.c]: thanks, fixed this and others

File: gdb/or1k-tdep.c (hunk 2 above)
Comment (stffrdhrn) [inline gdb/or1k-tdep.c]: thanks fixed this.
